Tasks/SeleniumWebDriverInstaller/Tests/L0.ts

```diff
@@ -0,0 +1,162 @@
+
+        // Set the inputs
+        process.env[constants.firefoxDriver] = 'true';
+        process.env[constants.firefoxDriverVersion] = '2.3.0';
+        process.env[constants.agentTempDirectory] = 'temp';
+        process.env[constants.downloadPath] = 'temp\\firefoxdriver';
+    
+        // Start the run
+        tr.run();
+
+        // Asserts
+        assert(tr.stderr.length === 0 || tr.errorIssues.length, 'should not have written to stderr');
+        assert(tr.succeeded, `Task should have succeeded`);
+        assert(tr.stdOutContained('Could not find version 2.3.0 of driver GeckoWebDriver, will download it'), 'Should have downloded firefox driver.');
+        assert(tr.stdOutContained('Setting Environment variable GeckoWebDriver=GeckoWebDriver\\2.3.0'));
+        done();
+    });
+
+    it('Get firefoxdriver acquires iedriver', (done: MochaDone) => {
+        console.log('TestCaseName: Get iedriver acquires iedriver');
+
+        // Setup the mock runner
+        const tp = path.join(__dirname, 'TestSetup.js');
+        const tr : ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        // Set the inputs
+        process.env[constants.ieDriver] = 'true';
+        process.env[constants.ieDriverVersion] = '2.3.0';
+        process.env[constants.agentTempDirectory] = 'temp';
+        process.env[constants.downloadPath] = 'temp\\iedriver';
+    
+        // Start the run
+        tr.run();
+
+        // Asserts
+        assert(tr.stderr.length === 0 || tr.errorIssues.length, 'should not have written to stderr');
+        assert(tr.succeeded, `Task should have succeeded`);
+        assert(tr.stdOutContained('Could not find version 2.3.0 of driver IEWebDriver, will download it'), 'Should have downloded ie driver.');
+        assert(tr.stdOutContained('Setting Environment variable IEWebDriver=IEWebDriver\\2.3.0'));
+        done();
+    });
+
+    it('Get edgedriver acquires edgedriver', (done: MochaDone) => {
+        console.log('TestCaseName: Get edgedriver acquires edgedriver');
+
+        // Setup the mock runner
+        const tp = path.join(__dirname, 'TestSetup.js');
+        const tr : ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+        // Set the inputs
+        process.env[constants.edgeDriver] = 'true';
+        process.env[constants.edgeDriverVersion] = '2.3.0';
+        process.env[constants.agentTempDirectory] = 'temp';
+        process.env[constants.downloadPath] = 'temp\\egdedriver';
+    
+        // Start the run
+        tr.run();
+
+        // Asserts
+        assert(tr.stderr.length === 0 || tr.errorIssues.length, 'should not have written to stderr');
+        assert(tr.succeeded, `Task should have succeeded`);
+        assert(tr.stdOutContained('Could not find version 2.3.0 of driver EdgeWebDriver, will download it'), 'Should have downloded edge driver.');
+        assert(tr.stdOutContained('Setting Environment variable EdgeWebDriver=EdgeWebDriver\\2.3.0'));
+        done();
+    });
+});
```

Comment:
add a test where it downloads more than one driver, and case where all pass and one of them fails. What is the expected behavior when one download fails but others pass? Do we want to go ahead with the other downloads and pass the task? We should ask prachi this